traffic_ops/testing/api/v4/cachegroups_parameters_test.go

```diff
@@ -79,6 +90,129 @@ func CreateTestCacheGroupParameters(t *testing.T) {
+	}
+	parameterID := *getAllcgparameters[0].Parameter
+
+	//Get cachegroup id from cachegroup name
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("name", *getAllcgparameters[0].CacheGroup)
+
+	cgResponse, _, _ := TOSession.GetCacheGroups(opts)
+	if len(cgResponse.Response) < 1 {
+		t.Fatalf("No data available for GetCacheGroups by Name - Length %d", len(cgResponse.Response))
+	}
+	if cgResponse.Response[0].ID == nil && *cgResponse.Response[0].ID != 0 {
+		t.Fatalf("found nil/0 Value in Cachegroup ID")
+	}
+	cachegroupID := *cgResponse.Response[0].ID
+	cgparameters, reqInf, err := TOSession.CreateCacheGroupParameter(cachegroupID, parameterID, client.RequestOptions{})
+	if reqInf.StatusCode != http.StatusBadRequest {
+		t.Errorf("Expected 400 status code, got %v", reqInf.StatusCode)
+	}
+	if err == nil {
+		t.Errorf("Expected Parameter - %d already associated with cachegroup %d - Alerts %v", parameterID, cachegroupID, cgparameters.Alerts)
+	}
+}
+
+func CreateTestCacheGroupParametersMulAssignments(t *testing.T) {
+	if len(testData.CacheGroups) < 3 || len(testData.Parameters) < 3 {
+		t.Fatal("Need at least three Cache Group and three Parameter to test associating Parameters to Cache Groups")
+	}
+	firstCacheGroup := testData.CacheGroups[1]
+	secondCacheGroup := testData.CacheGroups[2]
+	if firstCacheGroup.Name == nil {
+		t.Fatal("Found Cache Group1 with null or undefined name in test data")
+	}
+	if secondCacheGroup.Name == nil {
+		t.Fatal("Found Cache Group2 with null or undefined name in test data")
+	}
+
+	opts := client.NewRequestOptions()
+	opts.QueryParameters.Set("name", *firstCacheGroup.Name)
+	cacheGroupResp1, _, err := TOSession.GetCacheGroups(opts)
+	if err != nil {
+		t.Fatalf("cannot get Cache Group '%s': %v - alerts: %+v", *firstCacheGroup.Name, err, cacheGroupResp1.Alerts)
+	}
+	if len(cacheGroupResp1.Response) != 1 {
+		t.Fatalf("Expected exactly one Cache Group named '%s' to exist, but found %d", *firstCacheGroup.Name, len(cacheGroupResp1.Response))
+	}
+	opts.QueryParameters.Set("name", *secondCacheGroup.Name)
+	cacheGroupResp2, _, err := TOSession.GetCacheGroups(opts)
+	if err != nil {
+		t.Fatalf("cannot get Cache Group '%s': %v - alerts: %+v", *secondCacheGroup.Name, err, cacheGroupResp2.Alerts)
+	}
+	if len(cacheGroupResp2.Response) != 1 {
+		t.Fatalf("Expected exactly one Cache Group named '%s' to exist, but found %d", *secondCacheGroup.Name, len(cacheGroupResp2.Response))
+	}
+
+	// Get Parameter to assign to Cache Group
+	firstParameter := testData.Parameters[1]
+	secondParameter := testData.Parameters[2]
+	opts.QueryParameters.Set("name", firstParameter.Name)
+	paramResp1, _, err := TOSession.GetParameters(opts)
+	if err != nil {
+		t.Errorf("cannot get Parameter '%s': %v - alerts: %+v", firstParameter.Name, err, paramResp1.Alerts)
+	}
+	if len(paramResp1.Response) != 1 {
+		t.Fatalf("Expected exactly one Parameter named '%s' to exist, but found %d", firstParameter.Name, len(paramResp1.Response))
+	}
```

Comment:
Names don't uniquely identify Parameters. In fact, the second two Parameters in the testing data could easily have the same Name, they just coincidentally don't, currently. So you can't ever really expect only one Parameter to exist with a given Name without any other constraints.

Reply:
Fixed it.